app/assets/javascripts/discourse/components/pwa-install-banner.js.es6

```diff
@@ -0,0 +1,64 @@
+import {
+  default as computed,
+  on
+} from "ember-addons/ember-computed-decorators";
+
+const USER_DISMISSED_PROMPT_KEY = "dismissed-pwa-install-banner";
+
+export default Ember.Component.extend({
+  deferredInstallPromptEvent: null,
+
+  _handleInstallPromptEvent(event) {
+    // Prevent Chrome 76+ from automatically showing the prompt
+    event.preventDefault();
+    // Stash the event so it can be triggered later
+    this.set("deferredInstallPromptEvent", event);
+  },
+
+  @on("init")
```

Comment:
I agree it would be better in didInsertElement

Basic rule of thumb I use:
- any internal object state not directly impacting the DOM, setup in init
- anything with DOM, setup in didInsertElement